types/src/ledger_info.rs

```diff
@@ -376,9 +379,54 @@ impl LedgerInfoWithVerifiedSignatures {
     }
 }
 
-pub enum SignatureWithStatus {
-    Verified(bls12381::Signature),
-    Unverified(bls12381::Signature),
+#[derive(Clone, Debug, Derivative)]
+#[derivative(PartialEq, Eq)]
+pub struct SignatureWithStatus {
+    signature: bls12381::Signature,
+    #[derivative(PartialEq = "ignore")]
+    // false if the signature not verified.
+    // true if the signature is verified.
+    verification_status: Arc<AtomicBool>,
```

Comment:
Is `SignatureWithStatus` shared across threads? If yes, then use `Arc<SignatureWithStatus>` instead of `&SignatureWithStatus` everywhere?

Reply:
Even when we use `sig_with_status: &SignatureWithStatus` everywhere, and if some part of the code clones `sig_with_status`, then the cloned variable still refers to the same `verification_status` as we are using `Arc<AtomicBool>` right?

Reply:
Yes, it should be fine.